packages/workflow/src/internals.ts

```diff
@@ -20,14 +20,16 @@ import {
   WorkflowUpdateValidatorType,
   mapFromPayloads,
   fromPayloadsAtIndex,
+  WorkflowFunctionWithOptions,
+  VersioningBehavior,
 } from '@temporalio/common';
 import {
   decodeSearchAttributes,
   decodeTypedSearchAttributes,
 } from '@temporalio/common/lib/converter/payload-search-attributes';
 import { composeInterceptors } from '@temporalio/common/lib/interceptors';
 import { makeProtoEnumConverters } from '@temporalio/common/lib/internal-workflow';
-import type { coresdk, temporal } from '@temporalio/proto';
+import { coresdk, temporal } from '@temporalio/proto';
```

Comment:
Never do a regular import to `proto` from Workflow code (that's a strong no). And as a general rule, avoid using anything from `proto` in user-facing APIs (workflow code or not).

In this case, you will need to create a pure JS mirror of the enum. Search for `makeProtoEnumConverters` in the project for examples you can copy from. That JS type should be defined in the `common` package, as you will want it to also be accessible in non-workflow context.

Reply:
Thanks for the pointers!